kafka-client/src/main/java/dev/responsive/kafka/internal/db/WriteBatcher.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ *  * Copyright 2023 Responsive Computing, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package dev.responsive.kafka.internal.db;
+
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+import dev.responsive.kafka.internal.utils.Result;
+import java.util.Collection;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+public class WriteBatcher<K extends Comparable<K>, P> {
```

Comment:
As discussed offline, I think this makes sense to just be in the `CommitBuffer` - otherwise we can rename it something like `BatchFlusher` since it doesn't actually batch writes, it just flushes writes that have already been batched 😵‍💫 I'm tripping over words haha